lib/resty/rediscluster.lua

```diff
@@ -58,62 +86,135 @@ end
 _M.parse_key = parse_key
 
 
-local slot_cache = {}
-local master_nodes = {}
-
-local cmds_for_all_master = {
-    ["flushall"] = true,
-    ["flushdb"] = true
-}
-
-local cluster_invalid_cmds = {
-    ["config"] = true,
-    ["shutdown"] = true
-}
-
 local function redis_slot(str)
     return redis_crc(parse_key(str))
 end
 
-local function check_auth(self, redis_client)
-    -- redis before 6.0
-    if type(self.config.auth) == "string" then
+
+local function release_connection(red, config)
+    local ok,err = red:set_keepalive(config.keepalive_timeout
+            or DEFAULT_KEEPALIVE_TIMEOUT, config.keepalive_cons or DEFAULT_KEEPALIVE_CONS)
+    if not ok then
+        ngx_log(NGX_ERR, "set keepalive failed: ", err)
+    end
+end
+
+
+local check_auth
+do
+    local function check_reused(redis_client)
         local count, err = redis_client:get_reused_times()
-        if not count then
-            return nil, "failed getting reused count: " .. tostring(err)
-        elseif count > 0 then
-            return true, nil -- reusing the connection, so already authenticated
+        if err then
+            return nil, "failed to get reused count: " .. tostring(err)
         end
-        return redis_client:auth(self.config.auth)
 
-    -- redis 6.x adds support for username+password combination
-    elseif type(self.config.password) == "string" then
-        local count, err = redis_client:get_reused_times()
-        if not count then
-            return nil, "failed getting reused count: " .. tostring(err)
-        elseif count > 0 then
+        if count > 0 then
             return true, nil -- reusing the connection, so already authenticated
+        else
+            return nil, nil -- new connection
```

Comment:
the second return value both are nil, so that looks like a problem?

Reply:
Just a hint that this is a new connection. We can indeed discard it.